packages/maskbook/src/web3/UI/EthereumGasButton.tsx

```diff
@@ -0,0 +1,383 @@
+            borderRadius: 10,
+            margin: theme.spacing(1),
+            border: '1px solid grey',
+        },
+        title: {
+            display: 'flex',
+            flexDirection: 'row',
+            justifyContent: 'space-between',
+            alignItems: 'center',
+            paddingBottom: theme.spacing(1),
+        },
+        dot: {
+            width: 8,
+            height: 8,
+            borderRadius: 8,
+            backgroundColor: 'grey',
+        },
+        price: {
+            paddingBottom: theme.spacing(1),
+            flexDirection: 'row',
+            justifyContent: 'space-between',
+            display: 'flex',
+            alignItems: 'center',
+            height: 34,
+            '& > *: last-child': {
+                flex: 1,
+                paddingRight: theme.spacing(1),
+            },
+        },
+        eth: {
+            marginTop: theme.spacing(1),
+        },
+        text: {
+            borderRadius: 5,
+            marginRight: 5,
+            '& > *:first-child': {
+                borderRadius: 5,
+            },
+        },
+    })
+})
+
+interface GasPriceItemProps {
+    gasPrice: GasPrice
+    primary?: string
+    secondary?: string
+    active?: boolean
+    onChange?: (gasPrice: GasPrice) => void
+}
+function CalcETH(amount: string) {
+    return new BigNumber(amount).multipliedBy('0.00000002').multipliedBy(GAS_LIMIT).toFixed()
+}
+
+function GasPriceItem(props: GasPriceItemProps) {
+    const { gasPrice, primary = 'green', secondary = 'grey', active = false, onChange } = props
+    const [hover, setHover] = useState(false)
+    const [amountForUI, setAmountForUI] = useState(gasPrice.gasPrice)
+    const classes = useGasPriceItemStyles({ hover })
+    const [eth, setETH] = useState(CalcETH(gasPrice.gasPrice))
+    const [focuse, setFocuse] = useState(false)
+    const handleHoverIn = useCallback(() => {
+        setHover(true)
+        if (gasPrice.title === 'Custom') {
+            setFocuse(true)
+        }
+    }, [gasPrice.title])
+
+    const handleHoverOut = useCallback(() => {
+        setHover(false)
+        setFocuse(false)
+    }, [])
+
+    const handleAmount = useCallback(
+        (amount: string) => {
+            if (amount === '') setAmountForUI('')
+            if (/^\d+[\.]?\d*$/.test(amount)) {
+                setAmountForUI(amount)
+                setETH(CalcETH(amount || '0'))
+                onChange?.({
+                    ...gasPrice,
+                    gasPrice: amount,
+                })
+            }
+        },
+        [eth, gasPrice, onChange],
+    )
+
+    const onClick = useCallback(() => {
+        onChange?.(gasPrice)
+        if (gasPrice.title === 'Custom') {
+            setFocuse(true)
+        }
+    }, [gasPrice, onChange])
+
+    return (
+        <div
+            className={classes.root}
+            style={{
+                backgroundColor: hover ? secondary : 'inherit',
+                opacity: hover ? '0.6' : '1',
```

Comment:
Maybe you need `<Card />` with `<ButtonBase />` to make it behave like a button. I think `<Provider />` is a good example of this.